elisp/intero.el

```diff
@@ -490,10 +458,9 @@ CHECKER and BUFFER are added to each item parsed from STRING."
                        line column type
                        msg
                        :checker checker
-                       :buffer (when (string= (intero-buffer-file-name buffer)
-                                              file)
+                       :buffer (when (string= temp-file file)
                                  buffer)
-                       :filename file)
+                       :filename (intero-buffer-file-name buffer))
```

Comment:
We may want to also do a search-and-replace for the temp filename in error messages, which GHC sometimes references.